### ISSUE TITLE
Audit ODIS 2.0.0 testing

### ISSUE THREAD

**Issue by alecps** [celo-org/celo-monorepo#9811]
- [x] Make sure we're checking response values beyond just the status code everywhere
- [x] Make sure we're checking keyVersion header values in responses
- [x] Look for places where unit tests or further test cases would be useful
- [x] Make sure we're using strict object matching everywhere
- [x] Optionally reorganize the nesting of the combiner PNP/legacyPNP tests

The output of this ticket should be a PR and/or a sense of confidence that our testing is complete for the ODIS 2.0.0 refactor

**Comment by arthurgousset** [comment]
Sprint planning: 
- Catch all ticket, towards the end after other audit tickets
- Running code coverage testing **locally**
- Eela went for 3 dev days
- Alec went for 5 dev days: worried about unknown unknown
- Izzy went for 5 dev days: lot of thinking, unexpected things

Todo: 
- [ ] Alec - Some boxes currently don't make sense

**Comment by arthurgousset** [comment]
Sprint planning: 
- [ ] Alec split this up into combiner and signer tests

**Comment by arthurgousset** [comment]
Sprint check-in: 
- Eela - mostly done with checking combiner tests (added a couple tests), waiting on key rotation ticket for some tests
